File: Bai_94/main.cpp
#include <iostream>
#include <vector>
using namespace std;

int good_positions(int n, vector<int>& a) {
    int cnt = 0;
    for (int j = 0; j < n; ++j) {
        if (a[j] <= 0) continue;
        int sum = 0;
        bool good = true;
        for (int k = 0; k < n; ++k){
            sum += a[(j + k) % n];
            if (sum <= 0) {
                good = false;
                break;
            }
        }
        
        if (good) ++cnt;
    }
    return cnt;
}

int main() {
    int n;
    cin >> n;
    vector<int> a(n);
    for (int i = 0; i < n; ++i) 
        cin >> a[i];
    int result = good_positions(n, a);
    cout << result << endl;
    return 0;
}

File: Bai_94/main.py
def good_positions(n, a):
    cnt = 0
    for j in range(n):
        if a[j] <= 0: continue
        sum = 0
        good = True
        for k in range(n):
            sum += a[(j + k) % n]
            if sum <= 0:
                good = False
                break
        if good:
            cnt += 1
    
    return cnt

n = int(input())
a = list(map(int, input().split()))
res = good_positions(n, a)
print(res)